forte/processors/data_augment/ranking_augment_processor.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processors that augment the data.
+"""
+import random
+import string
+import nltk
+from nltk.tokenize.treebank import TreebankWordTokenizer, TreebankWordDetokenizer
+from typing import Tuple, List
+from forte.data.data_pack import DataPack
+from forte.data.multi_pack import MultiPack
+from forte.common.resources import Resources
+from forte.common.configuration import Config
+from forte.processors.base.data_augment_processor import ReplacementDataAugmentProcessor
+from ft.onto.base_ontology import (
+    Token, Sentence, Document, MultiPackLink
+)
+
+
+__all__ = [
+    "RankingDataAugmentProcessor"
+]
+
+random.seed(0)
+
+class RankingDataAugmentProcessor(ReplacementDataAugmentProcessor):
+    r"""
+    The data augmentation processor for ranking data.
+    The input multipack contains query pack (and document packs).
+    This class augments query and documents,
+    then adds the augmented datapacks to the original multipack.
+    """
+    def initialize(self, resources: Resources, configs: Config):
+        super().initialize(resources, configs)
+        self.tokenizer = TreebankWordTokenizer()
```

Comment:
We shouldn't call nltk here. we should use a NLTKProcessor beforehand and then run this.